packages/protocol/contracts/governance/DowntimeSlasher.sol

```diff
@@ -5,22 +5,33 @@ import "openzeppelin-solidity/contracts/math/SafeMath.sol";
 
 import "./SlasherUtil.sol";
 
+/// @dev Version: 2.0.0.0
 contract DowntimeSlasher is SlasherUtil {
   using SafeMath for uint256;
 
-  // For each address, associate each epoch with the last block that was slashed on that epoch
-  mapping(address => mapping(uint256 => uint256)) lastSlashedBlock;
+  // Maps validator address -> end block of the latest interval for which it has been slashed.
+  mapping(address => uint256) lastSlashedBlock;
+
+  // Maps validator -> startBlock -> endBlock -> signature bitmap for that interval.
+  // Note that startBlock and endBlock must always be in the same epoch.
+  mapping(address => mapping(uint256 => mapping(uint256 => bytes32))) private bitmaps;
```

Comment:
Why make this private? I would imagine it would be helpful to be able to view the contents here

Reply:
First of all, I realise that it's not "validator ->" in the comments, it's "user ->" (I'll change that)
I though this as a private, because I didn't want to have a user that could "read" for what another user paid to "compute".
And although it's difficult to have the address of the user, someone could know from an event that a specific address is "slashing" and only work using their previous validations.
Maybe I was over reacting, haha

Reply:
I'll change it